project/main.py

```diff
@@ -0,0 +1,111 @@
+    args = parser.parse_args()
+    return args
+
+
+
+
+
+def init_logger():
+    """
+    Here we will put potential
+    settings for the parser
+    :return:
+    """
+    logging.basicConfig(level=logging.DEBUG, format='%(asctime)s | %(levelname)s | %(pathname)s | %(message)s')
+    logging.getLogger('faker').setLevel(logging.ERROR)
+    return None
+
+
+
+
+
+def main():
+    """
+    Having main helps organize code,
+    we can see where main routine happens
+    We can clearly see the steps
+    :return:
+    """
+    print("Hello from main!")
+    init_logger()
+    parsed_args = init_arg_parser()
+    logging.info(parsed_args)
+    number_of_processes = os.cpu_count()
+
+    logging.info(f'Current path : {parsed_args.path_to_save_files}')
+    logging.info(f'You choose to generate {parsed_args.files_count} files')
+    logging.debug(f'Name of your files is {parsed_args.file_name} with prefixes {parsed_args.file_prefix}. '
+                  f'Every file have {parsed_args.data_lines} lines')
+
+    if parsed_args.path_to_save_files == ".":
+        path = os.getcwd()
+        logging.debug(f"You are in current directory. Path : {path}")
+    else:
+        path = utils.existing_dir(parsed_args.path_to_save_files)
+        logging.debug(f"Path : {path}")
+
+    if parsed_args.clear_path and parsed_args.path_to_save_files and parsed_args.file_name:
+        logging.info("You decide to clear path before work")
+        utils.clear_files_in_path(parsed_args.path_to_save_files, parsed_args.file_name)
+
+    if parsed_args.clear_path:
+        logging.info("Cleared data before generating files")
+
+    if int(parsed_args.multiprocessing) <= 0:
+        logging.error(f"You chose {parsed_args.multiprocessing} number of processes used to create files. Can't be less than 0")
+
+    if parsed_args.path_to_save_files is None and parsed_args.files_count is not None:
+        logging.info("You didn't choose the path dir, so path is current file ")
+
+    if int(parsed_args.files_count) > 0 and (parsed_args.file_name, parsed_args.file_prefix, parsed_args.path_to_save_files, parsed_args.data_lines, parsed_args.data_schema) is not None:
+        logging.debug(f'The number of processes used to create files is {parsed_args.multiprocessing}')
+        files = utils.construct_files(parsed_args.file_name, parsed_args.file_prefix, parsed_args.files_count)
+        workers_number = int(parsed_args.multiprocessing)
+        if workers_number > number_of_processes:
+            logging.info(f"Value of multiprocessing is higher than CPUs in the system. max CPU {number_of_processes} used instead")
+            workers_number = number_of_processes
+        with ThreadPoolExecutor(max_workers=workers_number) as executor:
+            futures = [executor.submit(utils.create_output_file, parsed_args.data_lines, parsed_args.data_schema, parsed_args.path_to_save_files, file) for file in files]
```

Comment:
Don't use list comprehensions for this because you are creating an additional python object with list in the memory. Using the for loop will be more efficient here.